main.go

```diff
@@ -259,15 +269,15 @@ func handleRequest(r *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http
 			proxyResp = resRecorder.Result()
 		} else {
 			// Set the Proxy URL and apply it to the Transpor layer so that the request respects the proxy.
-			proxyURL, _ := url.Parse("http://127.0.0.1:" + serverSettings.ExternalLegacyPort)
+			proxyURL, _ := url.Parse("http://127.0.0.1:" + strconv.Itoa(serverSettings.ExternalLegacyPort))
```

Comment:
The constant conversion in about a half dozen places is unnecessary, use the existing behavior of converting before assignment to serverSettings.